### DISCUSSION TITLE
How to reload UI after webhook event received

### DISCUSSION THREAD

**Discussion by sarimabbas** [remix-run/remix#2367] [Q&A] [answered]
Hello! I am new to Remix. I am currently trying to enqueue some long-running background task in a route action `/example`. The background task should complete and then hit my webhook at `/resources/task-completed`. Is there a way I can get the webhook to then reload the `/example` UI route? I noticed that we can use `useFetcher` for this on the client-side, but is there anything similar for the server-side? Alternatively is there a simpler way to queue background jobs and update UI when complete? 

I noticed https://github.com/remix-run/remix/pull/1363 and https://github.com/remix-run/remix/pull/2141 but it's not clear to me from the examples how the data loaders/UI would reload when the job completes and reports back. Thanks in advance to everyone for reading and their help!

**Comment by kiliman** [comment, marked as answer]
When the client submits a task, I'm assuming that task is added to a background process and you immediately return to the client.

At this point, the client won't know the status of the task. When the task is completed and notifies your webhook, the server doesn't have a way to let the client know it is completed.

There are a couple of ways to do this:

* Use long-polling on the client that will open a request to the server to get the status and wait until it has completed. This is relatively simple, but requires the client to remain on the status polling page.
* Use Web Sockets to notify the client when the status changes. Remix doesn't have built-in support for Web Sockets as this is part of the server process. There is a socket.io example using Express in the examples repo https://github.com/remix-run/remix/tree/main/examples/socket.io

**Reply by sarimabbas** [reply]
Awesome, makes sense! Thanks for your response :)
